compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java

```diff
@@ -104,7 +104,7 @@ public Symbol definition() {
             this.definition = symbolFactory.getBCompiledSymbol(((BParameterizedType) this.tSymbol.type).paramSymbol,
                                                                this.name());
         } else if (referredType.tag == TypeTags.INTERSECTION) {
-            this.definition = symbolFactory.getBCompiledSymbol(referredType.tsymbol,
+            this.definition = symbolFactory.getBCompiledSymbol(bType.tsymbol,
                     referredType.tsymbol.getName().getValue());
```

Comment:
What is the reason for this?

Reply:
Shouldn't we take the referred-type

Reply:
We should pass the `tsymbol` of the ref type right? If we do `referredType.tsymbol` we r passing the `tsymbol` of the intersection type

Reply:
Had a chat with @rdulmina live and we decided to go with this